frame/bridge/relay-authorities/src/lib.rs

```diff
@@ -793,9 +808,19 @@ where
 				});
 			}
 		} else {
-			if let Some(signatures) =
-				<MMRRootsToSign<T, I>>::take(now - <SubmitDuration<T, I>>::get())
-			{
+			let at = now - <SubmitDuration<T, I>>::get();
+
+			if let Some(signatures) = <MMRRootsToSign<T, I>>::take(at) {
+				let _ = <MMRRootsToSignKeys<T, I>>::try_mutate(|keys| {
+					if let Some(position) = keys.iter().position(|key| key == &at) {
+						keys.remove(position);
```

Comment:
Why need to remove the keys here ?

Reply:
We can't trace this data. Once the `SubmitDuration` changed.

```rust
let at = now.saturating_sub(<SubmitDuration<T, I>>::get());
<MMRRootsToSign<T, I>>::take(at);
```